### DISCUSSION TITLE
Would like to hide Side bar in mkdocs. Only on index page.

### DISCUSSION THREAD

**Discussion by jorgehcorrea** [mkdocs/mkdocs#2848] [Q&A]
Hi, 

I'm attempting to hide the toc in a mkdocs page. I have found the procedure in the squidfunk '`setup your own navegation'`(https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#back-to-top-button) documentation but I cannot get it to work, and its global. It would be most helpful to only have the hidden in one page. If I could just add a <style> tag on the top of the index.md it would be great. 

I have gotten through a couple of css suggested issues but I cannot get it to work as I am unsure what the class of the side bar is. 

Thanks in advance, 
cheers!

**Comment by skinitimski** [comment]
> I'm attempting to hide the toc in a mkdocs page. 

I wanted to do the same. I found that [the mkdocs website](https://www.mkdocs.org/) does this on its homepage. It is accomplished there with [this part of the "extra" css](https://github.com/mkdocs/mkdocs/blob/master/docs/css/extra.css#L41-L43).

Not a great answer in terms of discoverability, but it does work.

**Reply by ribafs** [reply]
Create a extra.css and enter:
.bs-sidebar{
    display: none;
}
In mkdocs.yml
extra_css:
  - extra.css
 
Here works

**Reply by brandonstrohmeyer** [reply]
This hides the sidebar on all pages. 

To hide the sidebar only on the homepage using extra.css:

```
body.homepage .bs-sidebar{
    display: none;
    }
```

**Comment by fu-sen** [comment]
I was dealing with this with [theme customization](https://www.mkdocs.org/user-guide/customizing-your-theme/).
For example, for `mkdocs` (same for `mkdocs-bootswatch`)
https://github.com/mkdocs/mkdocs/blob/3896a82ebf32a23cb63f53b1e84285ff7ddb2333/mkdocs/themes/mkdocs/base.html#L177-L180

Here we want to remove `toc.html`, Create the following as `main.html`:
```
{% extends "base.html" %}

{% block content %}
<div class="col-md-12" role="main">{% include "content.html" %}</div>
{% endblock %}
```

This method is a more complicated procedure than css, but it does not output html, so the merit is greater.
However, this will affect pages other than the index as well, so a little more tweaking is needed.

**Comment by RobertoPrevato** [comment]
Material for MkDocs' author recommends this approach to style single pages (I just came across this conversation and his comment):

https://github.com/squidfunk/mkdocs-material/issues/5941#issuecomment-1702785425
